### PR TITLE
Sort ESLint rules and fix Grunt scripts

### LINKED ISSUE #690: Remove a few stray JSCS references
Ref: #628

I removed a few unused `// jscs: ...` comments in #688, but doing some more greppin', it seems like we have a few other JSCS references buried in the dark corners of the code:

```sh
$ grep -irn "jscs" . --exclude-dir={node_modules,dist}

./.npmignore:6:.jscsrc
./CONTRIBUTING.md:15:Our tests include `eslint` and `jscs` style checks; these keep our code
./CONTRIBUTING.md:17:[`.jscsrc`][jscsrc] files.
./CONTRIBUTING.md:22:[jscsrc]: https://github.com/mozilla/addons-linter/blob/master/.jscsrc
./README.md:117:| grunt jscs             |  Checks for style issues  (Run in grunt test)    |
./tasks/jscs.js:4:      config: '.jscsrc',
```

I can do a PR for those references once #688 is merged.

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #690

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -17,36 +21,22 @@
...
   },
   "globals": {
-    "afterEach": false,
     "assert": false,
-    "beforeEach": false,
-    "describe": false,
-    "exports": false,
-    "it": false,
-    "module": false,
-    "require": false,
```
Comment (pdehaan) [inline .eslintrc]: All these removed globals were either in `env=node` (module, exports, require), or in `env=mocha` (afterEach, beforeEach, describe, it).

File: package.json [hunk 2]
```diff
@@ -7,11 +7,11 @@
     "addons-linter": "bin/addons-linter"
   },
   "scripts": {
-    "start": "node -e \"require('grunt').cli()\" null start",
```
Comment (muffinresearch) [inline package.json]: Does this now work if you don't have grunt-cli globally installed. That's the reason for this.

File: src/parsers/manifestjson.js [hunk 3]
```diff
@@ -109,7 +109,7 @@ export default class ManifestJSONParser {
       var id = this.parsedJSON.applications.gecko.id;
       return typeof id === 'undefined' ? null : id;
     } catch (e) {
-      console.log('Failed to get the id from the manifest.');
+      console.log('Failed to get the id from the manifest.'); // eslint-disable-line no-console
```
Comment (muffinresearch) [inline src/parsers/manifestjson.js]: This should use the logger instead. `log.error` should work.

File: tests/test.messages.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ describe('Messages', function() {
       it(`should not have any newlines in description for ${code}`, () => {
         assert.equal(description.split('\n').length, 1,
                      `The description for ${code} should not have newlines`);
-        console.log(description);
+        console.log(description); // eslint-disable-line no-console
```
Comment (muffinresearch) [inline tests/test.messages.js]: This one can be removed.

File: src/cli.js [hunk 5]
```diff
@@ -18,10 +18,8 @@ export function terminalWidth(_process=process) {
 }
 
 export default argv
-  // jscs:disable
```
Comment (muffinresearch) [inline src/cli.js]: Nice catch - jscs was removed a while back.

File: package.json (hunk 2 above)
Comment (pdehaan) [inline package.json]: It does (and passes Travis-CI). See http://gruntjs.com/blog/2016-02-11-grunt-1.0.0-rc1-released#npm-scripts

It took me a while to read the npm scripts and figure out what was going on with Grunt `cli()` and `null`s. Alternatively, we could have just done an `npm install grunt-cli -g` in the .travis.yml, but the above change in grunt@1 works really well.

File: src/cli.js (hunk 5 above)
Comment (pdehaan) [inline src/cli.js]: I just casually saw this, but we could/should probably do a grep and see if there are other buried `// jscs` comments lurking.

Comment (pdehaan) [inline src/cli.js]: Added another commit which removed all "jscs:" comments.

**Comment by muffinresearch** [comment]
If you can squash the commits this looks ready to land. Thanks!

**Comment by pdehaan** [comment]
Squashing.

**Pull request merged**